### ISSUE TITLE
npx react-native@latest init error installing ruby gems

### ISSUE THREAD

**Issue by maciejwiacek** [react-native-community/cli#2169]
While trying to create a react native CLI app on M1 Mac an error occured while installing Ruby Gems:

>✖ Installing Ruby Gems
error There was an error while trying to write to
/Users/maciejwiacek/.bundle/cache/compact_index/rubygems.org.443.29b0360b937aa4d161703e6160654e47/versions.61581.tmp.
It is likely that you need to grant write permissions for that path.
✖ Installing Ruby Gems
error Looks like your iOS environment is not properly set. Please go to https://reactnative.dev/docs/environment-setup?os=macos&platform=android and follow the React Native CLI QuickStart guide for macOS and iOS.

I tried reinstalling my macos, cocoapods, ruby versions, homebrew, watchman etc.

**Comment by thymikee** [comment]
The project is bootstrapped but the installation of Cocoapods failed. Can you run `npx react-native doctor` in the initialized app? Should give us some more information about your setup

**Comment by jasonk16** [comment]
> n

Mine seems to fail at `Installing CocoaPods dependencies` instead.  Here's the info when I run `npx react-native doctor`. I'm in the process of setting up the app for ios. 

```Common
 ✓ Node.js - Required to execute JavaScript code
 ✓ yarn - Required to install NPM dependencies
 ✓ Watchman - Used for watching changes in the filesystem when in development mode

Android
 ✖ Adb - No devices and/or emulators connected. Please create emulator with Android Studio or connect Android device.
 ✓ JDK - Required to compile Java code
 ✓ Android Studio - Required for building and installing your app on Android
 ✖ Android SDK - Required for building and installing your app on Android
   - Versions found: N/A
   - Version supported: 33.0.0
 ✖ ANDROID_HOME - Environment variable that points to your Android SDK installation

iOS
 ✓ Xcode - Required for building and installing your app on iOS
 ✓ Ruby - Required for installing iOS dependencies
 ✓ CocoaPods - Required for installing iOS dependencies
 ● ios-deploy - Required for installing your app on a physical device with the CLI
 ✖ .xcode.env - File to customize Xcode environment```

**Comment by maciejwiacek** [comment]
> The project is bootstrapped but the installation of Cocoapods failed. Can you run `npx react-native doctor` in the initialized app? Should give us some more information about your setup

I actually fixed it like couple minutes ago. The problem was that I was installing ruby with a help of sudo and it created permissions problems. After some guy told me to try and reinstall ruby once again but this time without sudo it worked. But thanks for help 😁

**Comment by jasonk16** [comment]
I managed to get it running when i downgraded back to yarn v1 instead of v4

**Comment by szymonrybczak** [comment]
> I managed to get it running when i downgraded back to yarn v1 instead of v4

Hm, interesting! What error did you have with Yarn v4?

**Comment by jasonk16** [comment]
> > I managed to get it running when i downgraded back to yarn v1 instead of v4
> 
> Hm, interesting! What error did you have with Yarn v4?

`Error: Cannot find module 'react-native/scripts/react_native_pods.rb'`, at the `Installing CocoaPods dependencies` stage of setting up the project.

**Comment by szymonrybczak** [comment]
@jasonk16 did you have `nodeLinker: node-modules` in your `.yarnrc.yml` config file? Because in React Native we don't support PnP.

**Comment by jasonk16** [comment]
> @jasonk16 did you have `nodeLinker: node-modules` in your `.yarnrc.yml` config file? Because in React Native we don't support PnP.

Oh that file is actually not there during setup. I ran `npx react-native@latest init AwesomeProject` and thats where it failed. I'll try to add it in and try it out again with yarn v4.

**Comment by szymonrybczak** [comment]
> Oh that file is actually not there during setup. 

Yeah, recently we fixed it recently. It will be released with 0.73 (in few weeks).

**Comment by jasonk16** [comment]
> > Oh that file is actually not there during setup.
> 
> Yeah, recently we fixed it recently. It will be released with 0.73 (in few weeks).

Oh gotcha thank you!
